frame/staking-rewards/src/lib.rs

```diff
@@ -946,3 +936,99 @@ pub mod pallet {
 		}
 	}
 }
+
+#[transactional]
+fn update_rewards_pool<T: Config>(
```

Comment:
What would be the advantage of having this as a separate function vs making it part of the pallet impl block?

Reply:
Needs a unit test

Reply:
Just personal preference, I like to keep the #[pallet] module as small as possible, and editing code outside of said macro makes the development cycle a lot faster since the proc macro doesn't have to run every time you make a change (also better ide support outside of the macro :sweat_smile:)

Reply:
I see, for my knowledge, is there any resources I could read up on to understand the development cycle time difference between those? 

In Clion I haven't seen any difference in IDE support between code with #[pallet] and not